### PR TITLE
allow adding plain files as release-assets

### PR DESCRIPTION
**Pull request by ccwienk**
Make it configurable whether release-assets should be wrapped into a tape-archive (aka tar). Also, improve documentation.

### CODE REVIEW

File: concourse/model/traits/release.py [hunk 1]
```diff
@@ -124,23 +173,6 @@ class BuildstepLogAsset(Asset):
     - type
     - target
```
Comment (8R0WNI3) [inline concourse/model/traits/release.py:174]: I think `target` is obsolete by now

File: concourse/model/traits/release.py [hunk 2]
```diff
@@ -124,23 +173,6 @@ class BuildstepLogAsset(Asset):
...
-
-    build
-    codecoverage
-    integrationtest
-    linter
-    sast          static code analysis results
-    test
-    unittest
 
     Purpose-labels will be added as OCM-Label:
```
Comment (8R0WNI3) [inline concourse/model/traits/release.py:177]: Maybe this should also be moved to base class docstring?

File: concourse/model/traits/release.py [hunk 3]
```diff
@@ -328,6 +360,9 @@ def deserialise(raw_asset: dict):
             return dacite.from_dict(
                 data_class=data_class,
                 data=raw_asset,
+                config=dacite.Config(
+                    cast=(enum.Enum,),
```
Comment (8R0WNI3) [inline concourse/model/traits/release.py:364]: Does this work actually? I think you have to cast to `FileAssetMode` 🤔

File: concourse/steps/release.mako [hunk 4]
```diff
@@ -107,6 +109,13 @@ import github.util
 import gitutil
 
 import git
+try:
+  import magic
+except ImportError:
+  # hack: install libmagic as workaround until libmagic will be installed into job-image
+  # -> remove this again
```
Comment (8R0WNI3) [inline concourse/steps/release.mako:116]: `libmagic` is installed into job-image by now -> https://github.com/gardener/cc-utils/commit/108702b5a0f7c9cedf09524bde863cd49a1aad0a

**Review by 8R0WNI3 — CHANGES_REQUESTED** [review]

File: concourse/model/traits/release.py (hunk 3 above)
Comment (ccwienk) [reply concourse/model/traits/release.py:364]: well, I tested it (and also pass enum.Enum at several other invocations of dacite.from_dict:

```
ccc/grafeas_model.py:129:                cast=[Enum],
concourse/model/traits/component_descriptor.py:285:                cast=(enum.Enum,),
concourse/model/traits/release.py:364:                    cast=(enum.Enum,),
concourse/steps/component_descriptor.mako:332:  config=dacite.Config(cast=(enum.Enum,)),
# ... (there are more, but I think those examples should do) :-)
```

File: concourse/steps/release.mako (hunk 4 above)
Comment (ccwienk) [reply concourse/steps/release.mako:116]: good catch - removed

File: concourse/model/traits/release.py (hunk 1 above)
Comment (ccwienk) [reply concourse/model/traits/release.py:174]: done

File: concourse/model/traits/release.py (hunk 3 above)
Comment (8R0WNI3) [reply concourse/model/traits/release.py:357]: hmm, and I wonder how dacite is able to find the correct enum class 🤔

Comment (ccwienk) [reply concourse/model/traits/release.py:357]: well that's easy, is it not? The concrete enum is added as typehint (as for all other types as well). To me, it is strange why dacite will not by default cast enums.

Consider the following code:
```
class E(enum.Enum):
 X = 'x'


@dataclasses.dataclass
class D:
 a: E
```

Then the following expression will create an instance of E:

```
D.__dataclass_fields__['a'].type('x')
```

-> as with all typehints for dataclass-attrs, dacite can simply use them (provided the values _are_ types) to "cast" to enum by invoking them.

So to my understanding, by passing `cast=[Enum]` to dacite, we merely tell it that it _should_ try such a cast. We do not tell it _how_ to cast (that we could to using type_hooks parameter). However, I do not quite understand _why_ dacite will not by default also try enum-casting.

**Review by 8R0WNI3 — APPROVED** [review]

**Pull request merged**